src/data/nat/digits.lean

```diff
@@ -281,6 +236,56 @@ begin
   { simp [of_digits, list.sum_cons, ih], }
 end
 
+/-! ### Inequalities
+
+This section contains various lemmas with inequalities relating to `digits` and `of_digits`.
+ -/
+
+/-- The digits in the base b+2 expansion of n are all less than b+2 -/
+lemma digits_lt_base' {b m : ℕ} : ∀ {d}, d ∈ digits (b+2) m → d < b+2 :=
+begin
+  apply nat.strong_induction_on m,
+  intros n IH d hd,
+  unfold digits at hd IH,
+  cases n with n,
+  { cases hd }, -- base b+2 expansion of 0 has no digits
+  rw digits_aux_def (b+2) (by linarith) n.succ (nat.zero_lt_succ n) at hd,
+  cases hd,
+  { rw hd, exact n.succ.mod_lt (by linarith) },
+  { exact IH _ (nat.div_lt_self (nat.succ_pos _) (by linarith)) hd }
+end
+
+/-- The digits in the base b expansion of n are all less than b, if b ≥ 2 -/
+lemma digits_lt_base {b m d : ℕ} (hb : 2 ≤ b) (hd : d ∈ digits b m) : d < b :=
+begin
+  rcases b with _ | _ | b; try {linarith},
+  exact digits_lt_base' hd,
+end
+
+/-- an n-digit number in base b + 2 is less than (b + 2)^n -/
+lemma of_digits_lt_base_pow_length' {b : ℕ} {l : list ℕ} (hl : ∀ x ∈ l, x < b+2) :
```

Comment:
same as above, I moved this from just under `of_digits`, but it can be at either location. I'm not sure which is the more logical place to put it.